File: api_nestjs_distributed_order_platform/src/application/use-case/client/delete-client.use-case.ts
import { Inject, Injectable, NotFoundException } from "@nestjs/common";
import type { ClientRepositoryInterface } from "src/application/port/client.repository";
import { CLIENT_REPOSITORY } from "src/shared/tokens_nest/client.token";

@Injectable()
export class DeleteClientUseCase {
    constructor(
        @Inject(CLIENT_REPOSITORY)
        private readonly clientRepository: ClientRepositoryInterface,
    ) {}

    async execute(clientId: string): Promise<void> {
        const client = await this.clientRepository.findById(clientId);
        if (!client) {
            throw new NotFoundException('Client not found');
        }

        await this.clientRepository.delete(clientId);
    }
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/application/usecase/client/DeleteClientUseCase.java
package com.project.order.application.usecase.client;

import com.project.order.application.exception.NotFoundException;
import com.project.order.application.port.ClientRepository;
import org.springframework.stereotype.Service;

@Service
public class DeleteClientUseCase {

    private final ClientRepository clientRepository;

    public DeleteClientUseCase(ClientRepository clientRepository) {
        this.clientRepository = clientRepository;
    }

    public void execute(String id) {
        clientRepository.findById(id)
                .orElseThrow(() -> new NotFoundException("Client not found"));
        clientRepository.delete(id);
    }
}
